data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlockRoot.java

```diff
@@ -0,0 +1,91 @@
+
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.PARAM_BLOCK_ID;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.PARAM_BLOCK_ID_DESCRIPTION;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_BAD_REQUEST;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_NOT_FOUND;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TAG_V1_BEACON;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_FOUND;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.Map;
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.beacon.GetBlockRootResponse;
+import tech.pegasys.teku.api.schema.Root;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetBlockRoot extends AbstractHandler implements Handler {
+  public static final String ROUTE = "/eth/v1/beacon/blocks/:block_id/root";
+  private final ChainDataProvider chainDataProvider;
+
+  public GetBlockRoot(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    this(dataProvider.getChainDataProvider(), jsonProvider);
+  }
+
+  public GetBlockRoot(final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get block root",
+      tags = {TAG_V1_BEACON},
+      description = "Retrieves hashTreeRoot of BeaconBlock/BeaconBlockHeader",
+      pathParams = {@OpenApiParam(name = PARAM_BLOCK_ID, description = PARAM_BLOCK_ID_DESCRIPTION)},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetBlockRootResponse.class)),
+        @OpenApiResponse(status = RES_BAD_REQUEST),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
```

Comment:
it'd be nice to show the endpoints actually create data that can deserialize. the easiest way is an integration test, and checking for an 'OK' response that the data block can be read.
also should check that block root not found gives a 404 in an integration test and maybe a future slot gives a bad request, and the bad request has a message.

Reply:
Done.